src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreBuildHandler.cs

```diff
@@ -117,6 +120,17 @@ public void Dispose()
 
         public void UpdateSolution_QueryDelayBuildAction(uint dwAction, out IVsTask pDelayTask)
         {
+            if (!_isMEFInitialized)
+            {
+                ThreadHelper.JoinableTaskFactory.Run(async () =>
```

Comment:
It seems like we're shifting the initialization cost elsewhere.  Are there opportunities to improve the initialization cost in general?